### PR TITLE
Avoid excessive IO in classfile parser

### PR DESCRIPTION
**Pull request by retronym**
The class file parser, used to read the java-defined classes from
the classpath, includes logic to search the output path for a .java
source file that corresponds to the value of the SourceFile attribute.

I haven't been able to figure out the rationale for that fix, but
it involves a non-neglible overhead, so this commits disables it
in the batch compiler.

### CODE REVIEW

**Comment by mkeskells** [comment]
@retronym Nice fix
slightly related - have you ever run with a security manager installed - logging all file opens would be a systematic way to spot all of the IO

**Review by lrytz — APPROVED** [review]

**Comment by retronym** [comment]
> slightly related - have you ever run with a security manager installed - logging all file opens would be a systematic way to spot all of the IO

@mkeskells I've used YourKit to identify the IO bottlenecks so far. This one was a bit of surprise!

**Comment by dragos** [comment]
@retronym I don't think that needs to stay in the IDE code path either. I think this is the ticket (at least, it links to the right commit on scalac): https://scala-ide-portfolio.assembla.com/spaces/scala-ide/tickets/realtime_cardwall?ticket=2689

File: src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala [hunk 1]
```diff
@@ -872,13 +872,26 @@ abstract class ClassfileParser {
...
+            // opt: disable this code in the batch compiler for performance reasons.
+            // it appears to be looking for the .java source file mentioned in this attribute
+            // in the output directories of scalac.
+            //
+            // References:
+            // https://issues.scala-lang.org/browse/SI-2689
+            // https://github.com/scala/scala/commit/7315339782f6e19ddd6199768352a91ef66eb27d
+            // https://github.com/scala-ide/scala-ide/commit/786ea5d4dc44065379a05eb3ac65d37f8948c05d
+            //
+            // TODO: can we disable this altogether? Does Scala-IDE actually intermingle source and classfiles in a way
```
Comment (fommil) [inline src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala:885]: we don't do anything funky in ensime that rings a bell here

**Pull request merged**